src/App.tsx

```diff
@@ -36,16 +35,16 @@ export default function App() {
         }
     })
 
-    // only use the api info's bbox until any other bounding box was chosen. Is this too messy?
-    const chooseBoundingBox = function (infoBbox: Bbox, shouldUseInfoBbox: boolean, pathBbox?: Bbox) {
-        if (shouldUseInfoBbox && pathBbox && pathBbox.every(num => num !== 0)) {
-            setUseInfoBbox(false)
-            return pathBbox
-        } else if (shouldUseInfoBbox) return infoBbox
-        return pathBbox || [0, 0, 0, 0]
-    }
-
-    const bbox = chooseBoundingBox(info.bbox, useInfoBbox, route.selectedPath.bbox)
+    const [bbox, setBbox] = useState<Bbox>([-180, -90, 180, 90])
+    useEffect(() => {
+        // make sure the path bbox takes precedence over the info bbox
+        if (!route.selectedPath.bbox)
+            setBbox(info.bbox)
+    }, [info])
+    useEffect(() => {
+        if (route.selectedPath.bbox)
+            setBbox(route.selectedPath.bbox)
+    }, [route])
```

Comment:
Using these two effects seems better than the previous code, because now we only update the bbox when the route or info changed, whereas previously we always had determine some bbox even when we did not want to change anything. This way we also no longer need the `shouldUseInfoBbox` flag.

Reply:
Yes, this is much better. Should we maybe combine those into a single effect though? Since you are using ```route``` in both effects maybe switch to something like
```
useEffect(() => setBbox(route.selectedPath.bbox ? route.selectedPath.bbox : info.bbox), [route, info])
```
I guess this is only a matter of taste though

Reply:
Ah yes this looks short and clean. But are you sure it does the same? Considering there were quite a few edge-cases (for which we have no tests currently :() and I was sure this version works and we might change this again in #87 I think I would rather leave it as is for this PR. Not the best reasons, I know...